ops/model.py

```diff
@@ -2211,6 +2211,7 @@ def __init__(
         self, name: str, backend: '_ModelBackend', pebble_client: Optional[pebble.Client] = None
     ):
         self.name = name
+        self._juju_context = backend._juju_context
```

Comment:
Ah, now I get it. I misunderstood it in the first place; I found that there was already a check on `supports_exec_service_context ` in `Container.exec`, so I thought it was already resolved.

How about storing `self.juju_version` in the init instead of just `self. supports_exec_service_context`? Because if the service context is not supported, the `RuntimeError` raised in `exec` would need to contain the version anyway. Or shall we remove the version from the warning?